packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -868,6 +887,104 @@ class ExactAssetImage extends AssetBundleImageProvider {
   String toString() => '${objectRuntimeType(this, 'ExactAssetImage')}(name: "$keyName", scale: $scale, bundle: $bundle)';
 }
 
+/// The action a [DeferringImageProvider] should take currently.
+enum DeferringImageProviderAction {
+  /// Indicates that the image is no longer needed, and the
+  /// [DeferringImageProvider] should stop seeking to resolve the image.
+  ///
+  /// After returning this action, the [DeferringImageProvider] will not call
+  /// its [DeferringImageProvider.getNextAction] callback anymore.
+  cancel,
+
+  /// Indicates that the caller would like to defer the image and be called
+  /// again after the next [DeferringImageProvider.deferDuration] elapses (or
+  /// at the end of current task queue if the duration is [Duration.zero]).
+  defer,
+
+  /// Indicates that the [DeferringImageProvider] should attempt to resolve the
+  /// image.
+  ///
+  /// After returning this action, the [DeferringImageProvider] will not call
+  /// its [DeferringImageProvider.getNextAction] callback anymore.
+  resolve,
+}
+
+/// An [ImageProvider] that can delay resolving an image using its wrapped
+/// [imageProvider] until an arbitrary condition is met.
+///
+/// For example, the [Image] widget uses this provider to defer loading images
+/// if it is the child of a scrollable that is rapidly scrolling. It cancels
+/// resolving the image if its element is evicted from the tree before scrolling
+/// ends, and otherwise waits for scrolling to slow down before actually
+/// resolving the image. This prevents the framework from over-utilizing
+/// hardware resources for images that will never be shown on screen.
+///
+/// If the image is already in the cache, e.g. from [precacheImage] or because
```

Comment:
add a paragraph above this one that explains how the provider decides wether to further defer or to resolve mentioning getNextAction.